pilot/pkg/xds/sds.go

```diff
@@ -304,24 +309,72 @@ func toEnvoyCaSecret(name string, cert []byte) *discovery.Resource {
 	}
 }
 
-func toEnvoyKeyCertSecret(name string, key, cert []byte) *discovery.Resource {
-	res := util.MessageToAny(&envoytls.Secret{
-		Name: name,
-		Type: &envoytls.Secret_TlsCertificate{
-			TlsCertificate: &envoytls.TlsCertificate{
-				CertificateChain: &core.DataSource{
-					Specifier: &core.DataSource_InlineBytes{
-						InlineBytes: cert,
+func fetchPrivateKeyProviderConfig(proxy *model.Proxy, configStore model.IstioConfigStore, meshConfig *mesh.MeshConfig) *mesh.PrivateKeyProvider {
+	if proxy == nil || meshConfig == nil || configStore == nil {
+		return nil
+	}
+	proxyConfs, err := model.GetProxyConfigs(configStore, meshConfig)
+	if err != nil {
+		return nil
+	}
+	if proxyConf := proxyConfs.EffectiveProxyConfig(proxy.Metadata, meshConfig); proxyConf != nil {
```

Comment:
@jacob-delgado  @Monkeyanator this doesn't seem to take `meta.ProxyConfig` into account which makes me suspicious that its not quite right. WDYT?

I mean the existing EffectiveProxyConfig code, not the new stuff

Reply:
Gentle ping @jacob-delgado @Monkeyanator

Reply:
Maybe we should be using ProxyConfigOrDefault?

EffectiveProxyConfig is used only for injection

Reply:
Yeah `ProxyConfigOrDefault` should be used here I think. `EffectiveProxyConfig` is used during injection which is where `meta.ProxyConfig` gets set from IIRC.

Reply:
Thanks for the comments, I have now modified to use default.

Reply:
PTAL @jacob-delgado